File: 프로그래머스/2/12900. 2 x n 타일링/2 x n 타일링.java
class Solution {
    public int solution(int n) {
        long[] D = new long[60001];
        D[1] = 1;
        D[2] = 2;
        for(int i = 3; i <= n; i++) {
            D[i] = (D[i-1] + D[i-2]) % 1000000007;
        }
        int answer = (int)D[n];
        return answer;
    }
}

File: 프로그래머스/2/12900. 2 x n 타일링/2 x n 타일링.js
function solution(n) {
    var answer = 0;
    const dp = Array(n+1).fill(0);
    if (n===1) return 1;
    if (n===2) return 2;
    dp[1] = 1;
    dp[2] = 2;
    for (let i = 3; i <= n; i++) {
        dp[i] = (dp[i-1] + dp[i-2]) % 1000000007;
    }
    return dp[n];
}